### PR TITLE
refactor: remove unused function and add comments in rc util

### PR DESCRIPTION
**Pull request by draveness**
*What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

https://github.com/kubernetes/kubernetes/issues/76206

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/framework/rc_util.go [hunk 1]
```diff
@@ -156,7 +115,7 @@ func DeleteRCAndWaitForGC(c clientset.Interface, ns, name string) error {
 	return DeleteResourceAndWaitForGC(c, api.Kind("ReplicationController"), ns, name)
 }
 
-// ScaleRC scales Replication Controller to be desired size.
+// ScaleRC scales replication controller to be desired size.
```
Comment (neolit123) [inline test/e2e/framework/rc_util.go]: Replication Controller should be upper case as in the official docs:
https://kubernetes.io/docs/tasks/run-application/rolling-update-replication-controller/

File: test/e2e/framework/rc_util.go [hunk 2]
```diff
@@ -188,9 +147,11 @@ func WaitForReplicationController(c clientset.Interface, namespace, name string,
 			Logf("Get ReplicationController %s in namespace %s failed (%v).", name, namespace, err)
 			return !exist, nil
 		}
+
```
Comment (neolit123) [inline test/e2e/framework/rc_util.go]: we can leave these whitespace cleanups out too.
[1]

File: test/e2e/framework/rc_util.go [hunk 3]
```diff
@@ -188,9 +147,11 @@ func WaitForReplicationController(c clientset.Interface, namespace, name string,
...
 		Logf("ReplicationController %s in namespace %s found.", name, namespace)
 		return exist, nil
 	})
+
```
Comment (neolit123) [inline test/e2e/framework/rc_util.go]: [1]

File: test/e2e/framework/rc_util.go [hunk 4]
```diff
@@ -82,46 +81,6 @@ func RcByNameContainer(name string, replicas int32, image string, labels map[str
 	}
 }
 
-// ScaleRCByLabels scales an RC via ns/label lookup. If replicas == 0 it waits till
-// none are running, otherwise it does what a synchronous scale operation would do.
-func ScaleRCByLabels(clientset clientset.Interface, scalesGetter scaleclient.ScalesGetter, ns string, l map[string]string, replicas uint) error {
```
Comment (neolit123) [inline test/e2e/framework/rc_util.go]: seems fine to be removed, IMO.

**Review by neolit123 — COMMENTED** [review]
/approve
/kind cleanup
/priority backlog

**Comment by draveness** [comment]
Hi, @neolit123  just updated, please take a look.

**Review by neolit123 — COMMENTED** [review]
/lgtm
/hold
for others to have a look too.

**Comment by andrewsykim** [comment]
/lgtm 

Couldn't find any callers for `ScaleRCByLabels`

**Pull request merged**